libs/designsystem/src/lib/components/segmented-control/segmented-control.component.spec.ts

```diff
@@ -68,6 +69,167 @@ describe('SegmentedControlComponent', () => {
+      });
+
+      it('should select focused item when Enter is pressed', async () => {
+        spectator.keyboard.pressKey('ArrowRight', ionSegmentElement, 'keydown');
+        const focusedItem = initSelectedSegmentButton.nextElementSibling;
+        expect(document.activeElement).toEqual(focusedItem);
+
+        spectator.keyboard.pressKey('Enter', ionSegmentElement, 'keydown');
+        // Wait for Ionic to update css classes
+        await TestHelper.whenTrue(() => focusedItem.classList.contains('segment-button-checked'));
+
+        expect(focusedItem).toHaveClass('segment-button-checked');
+      });
+    });
+    describe('with checked item', () => {
+      it('should have only 1 tab stop', async () => {
+        const segmentButtons =
+          spectator.queryHostAll<HTMLIonSegmentButtonElement>('ion-segment-button');
+
+        const segmentButtonsWithTabStop = segmentButtons.filter(
+          (button) => !button.hasAttribute('tabindex')
+        );
+        const segmentButtonsWithoutTabStop = segmentButtons.filter(
+          (button) => button.getAttribute('tabindex') === '-1'
+        );
+        expect(segmentButtonsWithTabStop).toHaveLength(1);
+        expect(segmentButtonsWithoutTabStop).toHaveLength(segmentButtons.length - 1);
+      });
+
+      it('should remove tab stop when focused', async () => {
+        const selectedSegmentButton = spectator.queryHost<HTMLIonSegmentButtonElement>(
+          'ion-segment-button.segment-button-checked'
+        );
+        expect(selectedSegmentButton).not.toHaveAttribute('tabindex');
+
+        await selectedSegmentButton.setFocus();
+        spectator.detectComponentChanges();
+
+        expect(selectedSegmentButton).toHaveAttribute('tabindex', '-1');
+      });
+
+      it('should add tab stop when blurred', async () => {
+        const selectedSegmentButton = spectator.queryHost<HTMLIonSegmentButtonElement>(
+          'ion-segment-button.segment-button-checked'
+        );
+        await selectedSegmentButton.setFocus();
+        spectator.detectComponentChanges();
+        expect(selectedSegmentButton).toHaveAttribute('tabindex', '-1');
+
+        selectedSegmentButton.blur();
+        spectator.detectComponentChanges();
+
+        expect(selectedSegmentButton).not.toHaveAttribute('tabindex');
+      });
+    });
+
+    describe('with no checked item', () => {
+      beforeEach(() => {
+        spectator.setInput('selectedIndex', -1);
+      });
+
+      it('should have only 1 tab stop', async () => {
+        const segmentButtons =
+          spectator.queryHostAll<HTMLIonSegmentButtonElement>('ion-segment-button');
+
+        const segmentButtonsWithTabStop = segmentButtons.filter(
+          (button) => !button.hasAttribute('tabindex')
+        );
+        const segmentButtonsWithoutTabStop = segmentButtons.filter(
+          (button) => button.getAttribute('tabindex') === '-1'
+        );
+        expect(segmentButtonsWithTabStop).toHaveLength(1);
```

Comment:
Would it be an idea to check that its the first segmentButton in the array that is without tabindex?